src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+type GetArea = () => number;
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: GetArea;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    readonly color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (!(this.a > 0) || !(this.b > 0) || !(this.c > 0)) {
```

Comment:
Simplify this condition

Reply:
I wrote this harder code to train himself read and write a bit difficult code. I'm not sure will I need it in future, hope not, but think better to be ready for it